docs/site/Validation-REST-layer.md

```diff
@@ -175,3 +175,76 @@ of 10 digits separated by `-` after the 3rd and 6th digits.
 ```
 
 {% include tip.html content="RegExp can be converted into a string with `.source` to avoid escaping backslashes" %}
+
+## Customize validation errors
+
+Since the error is being caught at the REST layer, the simplest way to customize
+the errors is to customize the
+[sequnce](https://loopback.io/doc/en/lb4/Sequence.html). It exists in all
+LoopBack applications scaffolded by using the `lb4` command and can be found in
+`src/sequence.ts`.
+
+Let's take a closer look at how to customize the error. A few things to note in
+the below code snippet:
+
+1. inject RestBindings.ERROR_WRITER_OPTIONS for logging error
+2. customize error for particular endpoints
+3. create a new error with customized properties
+4. log the error using RestBindings.SequenceActions.LOG_ERROR
+
+{% include code-caption.html content="/src/sequence.ts" %}
+
+```ts
+// 1. inject RestBindings.ERROR_WRITER_OPTIONS for logging error
+constructor(
+    @inject(SequenceActions.FIND_ROUTE) protected findRoute: FindRoute,
+    @inject(SequenceActions.PARSE_PARAMS) protected parseParams: ParseParams,
+    @inject(SequenceActions.INVOKE_METHOD) protected invoke: InvokeMethod,
+    @inject(SequenceActions.SEND) public send: Send,
+    @inject(SequenceActions.REJECT) public reject: Reject,
+    @inject(RestBindings.ERROR_WRITER_OPTIONS, {optional: true})
+    protected errorWriterOptions?: ErrorWriterOptions,
+  ) {}
+
+async handle(context: RequestContext) {
+    try {
+      //...
+    } catch (err) {
+      const httpError = <HttpErrors.HttpError>err;
+      // 2. customize error for particular endpoint
+      if (context.request.url === '/coffee-shops') {
+        // if this is a validation error
+        if (httpError.statusCode === 422) {
+          const customizedMessage = 'My customized validation error message';
+          const customizedProps = {
+            resolution: 'Contact your admin for troubleshooting.',
+            code: 'VALIDATION_FAILED',
+          };
+
+          // 3. Create a new error with customized properties
+          // you can change the status code here too
+          const newError: HttpErrors.HttpError = createHttpErrors(
```

Comment:
Do we really need a new error object? Please note that this new error will have a stack trace pointing to this line, not to the code that triggered the original error.

If we want to send this directly to HTTP response, then a plain data object should be good enough.

```ts
const errorData = {
  statusCode: 422,
  message: customizedMessage;
  ...customizedProps,
};

if (errorWriterOptions?.debug) {
  errorData.stack = httpError.stack;
}

context.response.status(422).send(newError);
```